src/concurrency/genmc/intercept.rs

```diff
@@ -9,64 +10,56 @@ use crate::{
 
 // Handling of code intercepted by Miri in GenMC mode, such as assume statement or `std::sync::Mutex`.
 
-impl<'tcx> EvalContextExt<'tcx> for crate::MiriInterpCx<'tcx> {}
-pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
-    /// Given a `ty::Instance<'tcx>`, do any required special handling.
-    /// Returns true if this `instance` should be skipped (i.e., no MIR should be executed for it).
-    fn check_genmc_intercept_function(
-        &mut self,
-        instance: rustc_middle::ty::Instance<'tcx>,
-        args: &[rustc_const_eval::interpret::FnArg<'tcx, crate::Provenance>],
-        dest: &crate::PlaceTy<'tcx>,
-    ) -> InterpResult<'tcx, bool> {
-        let this = self.eval_context_mut();
-        let genmc_ctx = this
-            .machine
-            .data_race
-            .as_genmc_ref()
-            .expect("This function should only be called in GenMC mode.");
-
-        let get_mutex_call_infos = || {
-            assert_eq!(args.len(), 1);
-            let arg = this.copy_fn_arg(&args[0]);
-            let addr = this.read_target_usize(&arg)?;
-            // FIXME(genmc): assert that we have at least 1 byte.
-            // FIXME(genmc): maybe use actual size of mutex here?.
-
-            let thread_infos = genmc_ctx.exec_state.thread_id_manager.borrow();
-            let curr_thread = this.machine.threads.active_thread();
-            let genmc_curr_thread = thread_infos.get_genmc_tid(curr_thread);
-            interp_ok((genmc_curr_thread, addr, 1))
-        };
+#[derive(Clone, Copy)]
+struct MutexMethodArgs {
```

Comment:
Why do we need a new *type* for this...??

Miri has hundreds of shim/intercept functions already. This should follow the same pattern as everything else, not make up something entirely new.

Reply:
I have pushed a commit adjusting the shims here to better match the rest of Miri. That also gets rid of this type.